plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataListing.java

```diff
@@ -103,15 +115,37 @@ public void tearDown()
     @Test
     public void testTableListing()
     {
-        assertQuery("SHOW TABLES FROM iceberg.test_schema", "VALUES 'iceberg_table1', 'iceberg_table2'");
+        assertThat(metastore.getAllTables("test_schema"))
+                .containsExactlyInAnyOrder(
+                        "iceberg_table1",
+                        "iceberg_table2",
+                        "iceberg_materialized_view",
+                        storageTable.getTableName(),
+                        "hive_table",
+                        "hive_view");
+
+        assertQuery(
+                "SHOW TABLES FROM iceberg.test_schema",
+                "VALUES " +
+                        "'iceberg_table1', " +
+                        "'iceberg_table2', " +
+                        "'iceberg_materialized_view', " +
+                        "'" + storageTable.getTableName() + "'");
     }
 
     @Test
     public void testTableColumnListing()
     {
         // Verify information_schema.columns does not include columns from non-Iceberg tables
-        assertQuery("SELECT table_name, column_name FROM iceberg.information_schema.columns WHERE table_schema = 'test_schema'",
-                "VALUES ('iceberg_table1', '_string'), ('iceberg_table1', '_integer'), ('iceberg_table2', '_double')");
+        // TODO this should include columns from the materialized view as well
```

Comment:
please create an issue in Github and reference it here

Reply:
This should get solved by https://github.com/trinodb/trino/pull/8113